ResidentInformationApi/V1/UseCase/ListContactsUseCase.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using ResidentInformationApi.V1.Boundary.Requests;
+using ResidentInformationApi.V1.Boundary.Responses;
+using ResidentInformationApi.V1.Factories;
+using ResidentInformationApi.V1.Gateways;
+
+namespace ResidentInformationApi.V1.UseCase
+{
+    public class ListContactsUseCase : IListContactsUseCase
+    {
+        private readonly IAcademyInformationGateway _academyGateway;
+        private readonly IHousingInformationGateway _housingGateway;
+        private readonly IMosaicInformationGateway _mosaicGateway;
+
+        public ListContactsUseCase(
+            IAcademyInformationGateway academyGateway,
+            IHousingInformationGateway housingGateway,
+            IMosaicInformationGateway mosaicGateway
+            )
+        {
+            _academyGateway = academyGateway;
+            _housingGateway = housingGateway;
+            _mosaicGateway = mosaicGateway;
+        }
+
+        public async Task<ResidentInformationResponse> Execute(ResidentQueryParam rqp)
+        {
+            var academyClaimants = await _academyGateway.GetClaimantInformation(rqp).ConfigureAwait(true);
+            var housingResidents = await _housingGateway.GetResidentInformation(rqp).ConfigureAwait(true);
+            var mosaicResidents = await _mosaicGateway.GetResidentInformation(rqp).ConfigureAwait(true);
+
+            var academyUrl = new Uri(Environment.GetEnvironmentVariable("ACADEMYAPI_URL"));
```

Comment:
Why are we using Uri objects instead of strings, do you know what a URI object looks like when it's converted to JSON in the API response? We want to make sure it is a string that is returned.

Reply:
Fair point, I can see that this is a string in the swagger dock, so I have updated the ResidentInformationResult object and the rest of the information

Reply:
There is a specific linting error which is demanding that `ResidentInformationResult.SystemUrl` should be `System.Uri` instead of `string`, but I can't tell why, unless it's down to a reading of the name. I've suppressed the error for the moment.

Reply:
Did you check what the JSON response was if you kept it as a Uri? It might not need to be changed to a string...